src/activities/bargame/bargame.js

```diff
@@ -0,0 +1,209 @@
+/* GCompris - bargame.js
+ *
+ * Copyright (C) 2016 UTKARSH TIWARI <iamutkarshtiwari@kde.org >
+ *
+ * Authors:
+ *   Yves Combe (GTK+ version)
+ *   UTKARSH TIWARI <iamutkarshtiwari@kde.org > (Qt Quick port)
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import GCompris 1.0 as GCompris
+.import QtQuick 2.0 as Quick
+
+var numberBalls = [[1, 4], [2, 6], [3, 6]]
+var boardSize = [15, 19, 29]
+var sampleBallsNumber = [4, 6, 6]
+var tuxPositionFactor = [1.05, 1.4, 1.05]
+var ballSizeFactor = [0, 4, 4]
+var elementSizeFactor = [0, 4, 7]
+var moveCount = -1
+var level = 1
+var maxlevel = 4
+var sublevel = 1
+var numberOfSublevel = 3
+var listWin = false
+var items
+var numberOfBalls = 1
+
+var url= "qrc:/gcompris/src/activities/bargame/resource/";
+
+function start(items_) {
+    items = items_
+    initLevel()
+}
+
+function stop() {
+}
+
+function initLevel() {
+    items.bar.level = level
+}
+
+function nextLevel() {
+    level++;
+    if (level > 4) {
+        level = 1;
+    }
+
+    reSetup();
+    initLevel();
+}
+
+function previousLevel() {
+    level--;
+    if (level < 1) {
+        level = 4;
+    }
+    sublevel = 1;
+
+    reSetup();
```

Comment:
In this activity, a lot of variable inside the js file are interdependent. Moving these variables to the qml file would need a lot of changes throughout the complete code which might introduce unprecedent bugs.